middleware/services/organizationcontainer.service.js

```diff
@@ -0,0 +1,49 @@
+const CONFIG = require('../config/config');
+const { ControlledContainerMixin, DisassemblyMixin } = require('@semapps/ldp');
+const {LDPNavigator,FetchAdapter} = require('fix-esm').require('ldp-navigator');
+const { defaultContext } = require('@semapps/core');
+const jsonld = require('jsonld');
+
+module.exports = {
+  name: 'organizations',
+  mixins: [ControlledContainerMixin, DisassemblyMixin],
+  settings: {
+    path: '/organizations',
+    acceptedTypes: ['pair:Organization'],
+    preferredView: '/Organization',
+    dereference: ['sec:publicKey', 'pair:hasLocation/pair:hasPostalAddress'],
+    disassembly: [{ path: 'pair:organizationOfMembership', container: CONFIG.HOME_URL + 'membership-associations' }],
+  },
+  actions: {},
+  hooks: {
+    after: {
+        "get": async (ctx, res) => {
+            let ldpNavigator=new LDPNavigator();
+            let headers = {
+              'accept': 'application/ld+json',
+            }
+            if (ctx.meta.headers && ctx.meta.headers.authorization){
+              headers.authorization = ctx.meta.headers.authorization
+            }
+            ldpNavigator.setAdapters([
+              new FetchAdapter({
+                headers : headers
+              })
+            ])
+            const oldContext= JSON.parse(JSON.stringify(res['@context']));
+            //context have to be replce because jsonld librairy don't support url with localhost
+            res['@context']=defaultContext['@context'];
+            await ldpNavigator.init(res);
+            res= await ldpNavigator.dereference(res, [{p:'pair:organizationOfMembership' }]);
+            console.log("tata")
+            //frame have to be because ldp-navigator return '@id' instead 'id' in dereferenced data
+            res = await jsonld.frame(res,{'@context':res['@context'],'id':ctx.params.resourceUri});
+            res['@context']=oldContext;
+
+            return res;
```

Comment:
Ce code est juste illisible. Peut-on le nettoyer, c'est-à-dire:
- Enlever le console.log
- Corriger les commentaires pour utiliser un anglais correct (utiliser deepl si besoin)
- Ajouter un commentaire global qui dit ce que ça fait, et aussi que c'est un "temporary hack".

D'ailleurs, est-ce que l'objet ldpNavigator ne devrait pas être initialisé seulement une fois, au départ du service, et non à chaque requête GET ?
S'il n'est pas prévu de se débarasser de ce "temporary hack" dans les mois qui viennent, je suggérerai que vous fassiez un mixin pour gérer ça. Il pourrait rester pour le moment au niveau d'Archipelago. Ca éviterait aussi la duplication du code. @simonLouvet

Reply:
Dans le code actuel, l'objet `ldpNavigator` est construit/configuré/initialisé à chaque appel de GET. On pourrait l'intialiser une fois dans la méthode `created` et l'utiliser dans le hook avec `this.ldpNavigator`.
Faire un mixin à partir de ces données prendrait moins de 30min. Je pense que ça devrait être fait dans cette PR.

Reply:
ok @srosset81, je comprends mieux. Ok sur le principe pour le créer/configurer une seul fois. L'initilisation c'est l'etape ou on lui donne des données de base sur lequel il va travailler qui sont différentes à chaque GET. Les différentes données initialisées vont rester en mémoire entre chaque appel.  Je ne sais pas comment va réagir ldpNavigator surtout si on fait 2 fois le même GET sur la même uri et des prédicats différents. Je crains qu'il n'écrase pas ce qu'il a déjà en mémoire (à vérifier). @BastienSig est ce que tu peux tenter?
Ok pour faire un mixin on hésitait.

Reply:
Ca vaudrait la peine de tester en effet...

Reply:
Salut @srosset81, j'ai bien réussi a créer la mixin, en effet c'était assez rapide à faire. Par contre pour initialiser le ldpNavigator dans un created, je suis bloqué par le header qui a besoin du ctx pour les autorisations ... Du coup je sais pas si tu as une idée de comment palier à ce problème ou si on valide la Pr comme ça.